python_modules/dagster/dagster/_core/executor/step_delegating/step_delegating_executor.py

```diff
@@ -222,6 +222,21 @@ def execute(self, plan_context: PlanOrchestrationContext, execution_plan: Execut
                             assert isinstance(dagster_event.step_key, str)
                             del running_steps[dagster_event.step_key]
                             active_execution.verify_complete(plan_context, dagster_event.step_key)
+                        if dagster_event.is_resource_init_failure:
+                            step_failure_event = DagsterEvent.step_failure_event(
+                                step_context=plan_context.for_step(
+                                    active_execution.get_step_by_key(dagster_event.step_key)
+                                ),
+                                step_failure_data=StepFailureData(
+                                    error=dagster_event.event_specific_data.error,  # type: ignore
+                                    user_failure_data=None,
+                                ),
+                            )
+                            yield step_failure_event
+                            active_execution.handle_event(step_failure_event)
+                            assert isinstance(dagster_event.step_key, str)
+                            del running_steps[dagster_event.step_key]
+                            active_execution.verify_complete(plan_context, dagster_event.step_key)
```

Comment:
something seems off if we have to dupe the logic in active exec here as well

Reply:
oops misread that above was multiproc not active ex, but something still fishy about this

Reply:
Agreed that it's not that clean. This seemed like the easiest solution; wanted to get takes on this approach first but another I have in mind would be having step start occur immediately after opening the process instead of waiting until initialization occurs first (I don't think that has any troublesome implications? And might just be cleaner).